eclass/distutils-r1.eclass

```diff
@@ -1495,40 +1486,22 @@ distutils-r1_python_compile() {
 			# call setup.py build when using setuptools (either via PEP517
 			# or in legacy mode)
 
-			if [[ ${DISTUTILS_USE_PEP517} ]]; then
-				if [[ -d build ]]; then
-					eqawarn "A 'build' directory exists already.  Artifacts from this directory may"
-					eqawarn "be picked up by setuptools when building for another interpreter."
-					eqawarn "Please remove this directory prior to building."
-				fi
-			else
-				_distutils-r1_copy_egg_info
-			fi
+			mkdir -p "${BUILD_DIR}" || die
+			local -x DIST_EXTRA_CONFIG="${BUILD_DIR}/extra-setup.cfg"
 
 			# distutils is parallel-capable since py3.5
-			local jobs=$(makeopts_jobs "${MAKEOPTS} ${*}")
+			cat > "${DIST_EXTRA_CONFIG}" <<-EOF
+			[build_ext]
+			parallel = $(makeopts_jobs "${MAKEOPTS} ${*}")
+			EOF
```

Comment:
Let's limit this new variable to PEP517 builds. We're explicitly calling `build -j` in legacy branch anyway.